cmd/root.go

```diff
@@ -79,98 +73,85 @@ func RunCmdRoot() {
 
 	targetOptions, err := opts.GetWarmupTargetOptions()
 
-	result := warmup.Result{RequestsToSend: 0, RequestsSent: 0}
 	if err == nil {
-		wp, err1 := createWarmup(targetOptions, done)
+		wp, err1 := createWarmup(targetOptions)
 		if err1 == nil {
-			result = runWarmup(wp, done)
-		}
-	}
-	log.Printf("summary: requests_to_send: %d, requests_sent: %d", result.RequestsToSend, result.RequestsSent)
-
-	if opts.ServerProbe.Enabled {
-		if opts.FailReadiness && result.RequestsSent == 0 {
-			log.Print("probeServer readiness failed, no requests sent")
+			runWarmup(wp)
+			log.Print("Warm up finished 😊")
 		} else {
-			probeServer.IsReady(true)
+			log.Print("Target still not ready. Giving up! No requests were sent 🙁")
 		}
-	}
-	if opts.FileProbe.Enabled {
-		if opts.FailReadiness && result.RequestsSent == 0 {
-			log.Print("fileServer readiness failed, no requests sent")
+
+		if opts.FailReadiness && err1 != nil {
```

Comment:
I'm not sure if replacing the current logic for failing readiness with this is a good idea. This is purely based on `err1 ` which only takes into account the (successful) creation of the warmup (line 77).

There are many places where things can go wrong before sending the actual request.
- When we [construct the headers](https://github.com/ExpediaGroup/mittens/blob/0be492fbe56a9438a57a323a1a50af461948d5a0/cmd/root.go#L126)
- When we [create the request](https://github.com/ExpediaGroup/mittens/blob/4ff7a489ca072e00bbbc069842bf78e014a72dca/pkg/http/client.go#L60)
- When we send the request and this results to an [error](https://github.com/ExpediaGroup/mittens/blob/master/pkg/http/client.go#L74) (without this meaning that it's a non-200)

Note that with the current logic requests that returned a non-200 are still [treated as successful ones](https://github.com/ExpediaGroup/mittens/blob/master/pkg/http/client.go#L83) in the context of this flag. We only treat it as a failure when there's an error as this is defined by [Go's HTTP client](https://github.com/golang/go/blob/master/src/net/http/client.go#L552-L555).

Imho unless we base the `failReadiness` flag's implementation to this internal information the result will be misleading.

Let me know your thoughts.

Reply:
Right, I now see how you define the `-fail-readiness` flag in the docs which is "target did not become ready in time ". Above comment still applies but it'd be good to discuss if this definition would provide value to users. :)

Reply:
agree. updated.